scripts/run-e2e-tests.sh

```diff
@@ -101,14 +101,24 @@ function install_kind() {
 }
 
 function install_spark() {
-  check_cmd "wget"
-  wget -qO-  https://archive.apache.org/dist/spark/spark-3.3.1/spark-3.3.1-bin-hadoop3.tgz | tar xzvf - && chmod +x spark-3.3.1-bin-hadoop3 && sudo mv spark-3.3.1-bin-hadoop3 /opt/.
-  exit_on_error "install spark failed."
-  sudo ln -s /opt/spark-3.3.1-bin-hadoop3 /opt/spark
-  exit_on_error "install spark failed. unable to create symlink"
-  export SPARK_HOME=/opt/spark/
-  export SPARK_PYTHON_IMAGE=docker.io/apache/spark-py:v3.3.1
+  CURRENT=$(pwd)
+  SPARK_VERSION="3.3.1"
+  DOWNLOAD="${CURRENT}/_spark"
+  SPARK_SUBMIT_CMD="${DOWNLOAD}/spark/bin/spark-submit"
+  if [[ ! -d "${DOWNLOAD}" || ! -f "${SPARK_SUBMIT_CMD}" ]]; then
+    echo  "Installing spark in yunikorn rep."
+    rm -rf "${DOWNLOAD}"
+    mkdir "${DOWNLOAD}" \
+      && cd "${DOWNLOAD}" \
+      && curl "https://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.tgz" | tar xzvf - \
+      && mv "spark-${SPARK_VERSION}-bin-hadoop3" spark \
+      && chmod +x spark
+    exit_on_error "install spark failed."
+  fi
+  export SPARK_HOME="${DOWNLOAD}/spark"
```

Comment:
Set this to "${DOWNLOAD}/spark-${SPARK_VERSION}-bin-hadoop3"

Reply:
Ok